packages/caliper-core/lib/worker/tx-observers/prometheus-push-tx-observer.js

```diff
@@ -81,27 +119,56 @@ class PrometheusPushTxObserver extends TxObserverInterface {
      */
     async activate(roundIndex, roundLabel) {
         await super.activate(roundIndex, roundLabel);
-        this.intervalObject = setInterval(async () => { await this._sendUpdate(); }, this.sendInterval);
+
+        // update worker and round metadata
+        this.defaultLabels.workerIndex = this.workerIndex;
+        this.defaultLabels.roundIndex = this.currentRound;
+        this.defaultLabels.roundLabel = this.roundLabel;
+        this.registry.setDefaultLabels(this.defaultLabels);
+
+        const self = this;
```

Comment:
Is self necessary ? I thought javascript correctly handled it when using arrow functions